src/minesweeper/infrastructure/controllers/flags/post/flags.post.controller.ts

```diff
@@ -0,0 +1,67 @@
+import { dataRepository } from '@minesweeper/infrastructure/data';
+import { RequestListener } from 'http';
+import { ServerError } from '@minesweeper/infrastructure/server/utils/validations';
+import getBody from '@minesweeper/infrastructure/server/utils/getBody';
+import { markFlagUseCase } from '@minesweeper/use-cases/markFlag/markFlag';
+import { TPosition } from '@minesweeper/domain/models';
+
+export enum flagsPostControllerErrors {
+  EMPTY_BODY = 'must send position and boardId',
+  BOARD_ID_IS_REQUIRED = 'board id is required',
+  BOARD_ID_MUST_BE_STRING = 'board id must be a string',
+  POSITION_IS_REQUIRED = 'position is required',
+  POSITION_MUST_BE_ARRAY = 'position must be a array',
+  POSITION_MUST_BE_ARRAY_OF_NUMBERS = 'position must be a array of two numbers',
+}
+
+const getFlagsParameters = (
+  body: Record<string, unknown>,
+): { boardId: string; position: TPosition } => {
+  const boardEntry = Object.entries(body).find(([key]) => key === 'boardId');
+
+  if (!boardEntry) throw new ServerError(400, flagsPostControllerErrors.BOARD_ID_IS_REQUIRED);
+
+  const boardId = boardEntry[1];
+
+  if (typeof boardId !== 'string')
+    throw new ServerError(400, flagsPostControllerErrors.BOARD_ID_MUST_BE_STRING);
+
+  const positionEntry = Object.entries(body).find(([key]) => key === 'position');
+
+  if (!positionEntry) throw new ServerError(400, flagsPostControllerErrors.POSITION_IS_REQUIRED);
+
+  const position = positionEntry[1];
```

Comment:
I prefer to type this instead of having an any or unknown but i understand why this approach, is more "the typescript inference way"